src/node_messaging.cc

```diff
@@ -795,6 +795,10 @@ void MessagePort::OnMessage(MessageProcessingMode mode) {
 
   size_t processing_limit;
   if (mode == MessageProcessingMode::kNormalOperation) {
+    // Maybe the async handle was triggered empty or more than needed
+    // (called twiced on a new context).
+    if (IsDetached()) return;
```

Comment:
It’s probably fine to move this to the very top of the function, right? It would also be great if the comment here could describe the sequence of events that leads to e.g. `data_` being null.